libiocage/lib/JailConfig.py

```diff
@@ -158,34 +128,7 @@ def clone(self, data, skip_on_error=False):
             self.__setitem__(key, value, skip_on_error=skip_on_error)
 
     def read(self):
-
-        if libiocage.lib.JailConfigJSON.JailConfigJSON.exists(self):
-
-            libiocage.lib.JailConfigJSON.JailConfigJSON.read(self)
-            self.legacy = False
-            self.logger.log("Configuration loaded from JSON", level="verbose")
-            return "json"
-
-        elif libiocage.lib.JailConfigLegacy.JailConfigLegacy.exists(self):
-
-            libiocage.lib.JailConfigLegacy.JailConfigLegacy.read(self)
-            self.legacy = "ucl"
-            self.logger.verbose(
-                "Configuration loaded from UCL config file (iocage-legacy)")
-            return "ucl"
-
-        elif libiocage.lib.JailConfigZFS.JailConfigZFS.exists(self):
-
-            libiocage.lib.JailConfigZFS.JailConfigZFS.read(self)
-            self.legacy = "zfs"
-            self.logger.verbose(
-                "Configuration loaded from ZFS properties (iocage-legacy)")
-            return "zfs"
-
-        else:
-
-            self.logger.debug("No configuration was found")
-            return None
+        self.clone(self.jail.resource.read_config())
```

Comment:
This doesn't return anything anymore?

Reply:
No, this read function calls clone on it's own. So it automatically adapts the loaded configuration. I'll make a note to even out all methods in this project with the same name behave in the same way!